src/Altinn.Apps/AppTemplates/AspNet/App.IntegrationTests/ApiTests/InstanceApiTest.cs

```diff
@@ -341,5 +324,110 @@ public async Task Instance_Post_WithLowAuthLevel_FailOk()
+        [Fact]
+        public async void AddCompleteConfirmation_PostAsValidAppOwner_RespondsWithUpdatedInstance()
+        {
+            // Arrange
+            string org = "tdd";
+            string app = "endring-av-navn";
+            int instanceOwnerPartyId = 1337;
+            string instanceGuid = "66233fb5-a9f2-45d4-90b1-f6d93ad40713";
+
+            TestDataUtil.DeleteInstanceAndData(org, app, instanceOwnerPartyId, new Guid(instanceGuid));
+            TestDataUtil.PrepareInstance(org, app, instanceOwnerPartyId, new Guid(instanceGuid));
+        
+            HttpClient client = SetupUtil.GetTestClient(_factory, org, app);
+
+            string token = PrincipalUtil.GetOrgToken(org);
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+
+            string requestUri = $"/{org}/{app}/instances/{instanceOwnerPartyId}/{instanceGuid}/complete";
+
+            // Act
+            HttpResponseMessage response = await client.PostAsync(requestUri, new StringContent(string.Empty));
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            string json = await response.Content.ReadAsStringAsync();
+            Instance updatedInstance = JsonConvert.DeserializeObject<Instance>(json);
+
+            // Don't compare original and updated instance in asserts. The two instances are identical.
+            Assert.NotNull(updatedInstance);
+            Assert.Equal(org, updatedInstance.CompleteConfirmations[0].StakeholderId);
+            Assert.Equal($"{instanceOwnerPartyId}/{instanceGuid}", updatedInstance.Id);
+        }
+
+        /// <summary>
+        /// Scenario:
+        ///   An Application Owner attempts to complete an instance for an application not owned by them.
+        /// Result:
+        ///   The response returns status code Forbidden.
+        /// </summary>
+        [Fact]
+        public async void AddCompleteConfirmation_AttemptToCompleteInstanceAsDifferentOrg_ReturnsForbidden()
+        {
+            // Arrange
+            string org = "tdd";
+            string app = "endring-av-navn";
+            int instanceOwnerPartyId = 1337;
+            string instanceGuid = "66233fb5-a9f2-45d4-90b1-f6d93ad40713";
+
+            TestDataUtil.DeleteInstanceAndData(org, app, instanceOwnerPartyId, new Guid(instanceGuid));
+            TestDataUtil.PrepareInstance(org, app, instanceOwnerPartyId, new Guid(instanceGuid));
+            
+            HttpClient client = SetupUtil.GetTestClient(_factory, org, app);
+
+            string token = PrincipalUtil.GetOrgToken("brg");
+            client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Bearer", token);
+
+            string requestUri = $"/{org}/{app}/instances/{instanceOwnerPartyId}/{instanceGuid}/complete";
+
+            // Act
+            HttpResponseMessage response = await client.PostAsync(requestUri, new StringContent(string.Empty));
+
```

Comment:
Could we include clean up of the testdata if the test results in modifications to the testdata? 
Haven't run the tests, so not sure if this is the case or not.

Reply:
I hope I got it right, but I have random results so I don't feel too confident. Taking the chance.